task/mgmt-gateway/src/mgs_handler.rs

```diff
@@ -0,0 +1,130 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use crate::Log;
+use crate::MgsMessage;
+use crate::UsartHandler;
+use crate::__RINGBUF;
+use gateway_messages::sp_impl::SocketAddrV6;
+use gateway_messages::sp_impl::SpHandler;
+use gateway_messages::BulkIgnitionState;
+use gateway_messages::DiscoverResponse;
+use gateway_messages::IgnitionCommand;
+use gateway_messages::IgnitionState;
+use gateway_messages::ResponseError;
+use gateway_messages::SerialConsole;
+use gateway_messages::SpPort;
+use gateway_messages::SpState;
+use ringbuf::ringbuf_entry;
+
+pub(crate) struct MgsHandler<'a> {
```

Comment:
I notice that you're creating and destroying these on every use. Is that deliberate? Should it instead own the `usart` (rather than holding a reference) and persist across operations?

Reply:
I vaguely remember trying this and running into a lifetime issue, but don't remember the details. I'll try this again and either succeed or post a followup.

Reply:
:man_shrugging: don't know what I ran into before; 7dc7425b makes this owned instead of borrowed and avoids creating/destroying these on the fly.